File: tests/fixtures/sample.ts
export interface Greeter {
  greet(name: string): string;
}

export class FriendlyGreeter implements Greeter {
  constructor(private readonly prefix: string = "Hello") {}

  greet(name: string): string {
    return `${this.prefix}, ${name}!`;
  }
}

const greeter = new FriendlyGreeter();
const names = ["alice", "bob", "carol"];
for (const name of names) {
  console.log(greeter.greet(name));
}

File: tests/fixtures/sample.rs
use std::collections::HashMap;

/// Greet the world. The classic.
pub fn greet(name: &str) -> String {
    format!("Hello, {name}!")
}

pub struct Counter {
    count: u32,
    label: String,
}

impl Counter {
    pub fn new(label: &str) -> Self {
        Self {
            count: 0,
            label: label.to_string(),
        }
    }

    pub fn tick(&mut self) -> u32 {
        self.count += 1;
        self.count
    }
}

fn main() {
    let mut counts: HashMap<String, u32> = HashMap::new();
    let names = ["alice", "bob", "carol"];
    for name in names {
        *counts.entry(name.to_string()).or_insert(0) += 1;
        println!("{}", greet(name));
    }
}
